### PR TITLE
Integrating the Signer and adding Cypress tests

### PR DESCRIPTION
**Pull request by George-cl**
### Overview

Adds automated E2E testing for Clarity and Signer. Will need to be fleshed out a bit in future and it would be good to go through Clarity adding Cypress tags to elements to avoid breaking tests if DOM layout changes.
Cypress doesn't support testing extensions in the same way it does websites. As a result of this I wasn't able to interact with the Signer UI and instead opted to expose/create RPC methods to manipulate it programatically.
Some of these methods could pose security risks so I extracted them to a separate file and have changed the packaging config to ignore it in the build.
The result is that the testing methods are available via the window object when running it locally/in dev but it is omitted from the packaged .zip file.

This PR also merges the Signer into the packages of Clarity.

Any questions let me know.

### Which JIRA ticket does this PR relate to?

[ECO-783](https://casperlabs.atlassian.net/browse/ECO-783?atlOrigin=eyJpIjoiZWQyMTExZTM5MWZkNDIzZmJmOWY3MGJiYTVjZjYzYzciLCJwIjoiaiJ9),
[ECO-809](https://casperlabs.atlassian.net/browse/ECO-809?atlOrigin=eyJpIjoiZGZlNTYyOWMxZGQ0NDQwYjk0N2I0M2E5ZGY1NjAxMGEiLCJwIjoiaiJ9),
[ECO-808](https://casperlabs.atlassian.net/browse/ECO-808?atlOrigin=eyJpIjoiMzA5N2ZhNTM3NjNkNDFkMzkyOWFmYWI5N2Q1YTFiMGMiLCJwIjoiaiJ9)

### Complete this checklist before you submit this PR

- [ ] This PR contains no more than 200 lines of code, excluding test code.
- [ ] This PR meets [CasperLabs coding standards](https://casperlabs.atlassian.net/wiki/spaces/EN/pages/16842753/Coding+Standards).
- [ ] If this PR adds a new feature, it includes tests related to this feature.
- [ ] You assigned one person to review this PR.
- [ ] Your GitHub account is linked with our [Drone CI](https://drone-auto.casperlabs.io/) system. This is necessary to run tests on this PR.
- [ ] Do not forget to run `bors r+` if GitHub policy is not enforced, e.g. when merging into another feature branch. It may be omitted under some circumstances if this PR intentionally assumes that integration tests will fail but will be fixed with the future PRs.

### CODE REVIEW

**Comment by George-cl** [comment]
To test if it works:  
Spin up a local network:  
```
nctl-assets-setup && nctl-start
```
Pull down the code, install dependencies, build the packages and run Clarity locally:
```
git clone git@github.com:CasperLabs/clarity
cd clarity
git checkout George/Signer-Cypress-Integration
yarn install
yarn build
yarn dev
```
In another terminal, run the Cypress tests:
```
yarn cypress:chrome
```

**Comment by George-cl** [comment]
This PR is still in draft because I am adding further tests to the suite

**Comment by George-cl** [comment]
New tests added, and I created two distinct contexts to organise tests involving the Signer and those solely reliant on Clarity.

To check out Cypress without it closing as soon as the tests finish, run the command;
```
node_modules/.bin/cypress open
```

**Comment by George-cl** [comment]
I have removed the work for [OP-1746](https://casperlabs.atlassian.net/browse/OP-1746?atlOrigin=eyJpIjoiN2ViOGE2OTQ2NDQ1NGY4NGEyNWM2ZTJmOTUxMWVkOWUiLCJwIjoiaiJ9) as it was causing a strange regression which I couldn't immediately see a solution to.
I would prefer it not hold up this entire PR for the sake of one small ticket so I'm proposing that we merge this and then I'll update 'axios' (and I think a few of the other outdated/deprecated packages) at a later date.

File: packages/sdk/tslint.json
Comment (zie1ony) [inline packages/sdk/tslint.json]: Why is this removed?

File: packages/ui/src/services/AuthService.ts
Comment (zie1ony) [inline packages/ui/src/services/AuthService.ts:28]: I had the same issue, but @AbnerZheng managed to fix it. Do you remember that?

**Review by zie1ony — CHANGES_REQUESTED** [review]
Looks good

File: packages/sdk/tslint.json
Comment (George-cl) [reply packages/sdk/tslint.json]: Screenshot from 2021-01-08 16-00-17

Looks like it changed after you merged this branch with 'master'. I can add `"tslint-config-prettier"` back in if you want? @zie1ony

Comment (zie1ony) [reply packages/sdk/tslint.json]: So my bad. Yes, lets bring it back.

File: packages/ui/src/services/AuthService.ts
Comment (George-cl) [reply packages/ui/src/services/AuthService.ts:28]: @zie1ony / @AbnerZheng - does this need changed urgently or could we merge and then fix later (if needs be)?  

Screenshot from 2021-01-08 16-03-48

File: packages/sdk/tslint.json
Comment (George-cl) [reply packages/sdk/tslint.json]: https://github.com/CasperLabs/clarity/pull/81/commits/6aa79b4d6fe2b5722a2407fd76062bf3cf750c7d - done

File: packages/ui/src/services/AuthService.ts
Comment (AbnerZheng) [reply packages/ui/src/services/AuthService.ts:28]: No, we can do it later. I have found you merge the master and resolve conflict several times. Let's merge it firstly.

Comment (AbnerZheng) [reply packages/ui/src/services/AuthService.ts:28]: No, we can do it later. I have found you merge the master and resolve conflict several times. Let's merge it firstly.

**Review by AbnerZheng — APPROVED** [review]
Fantastic tool for testing a browser plugin.

File: packages/ui/src/services/AuthService.ts
Comment (George-cl) [reply packages/ui/src/services/AuthService.ts:28]: Sounds good

**Review by zie1ony — APPROVED** [review]

**Pull request merged**
